acceptance/analyzer_test.go

```diff
@@ -145,34 +127,57 @@ func setupAnalyzeFixtures(t *testing.T) analyzeFixtures {
 
 	// Registry
 
+	// setup read and write images
 	someAppName := "some-app-image-" + h.RandString(10)
 	fixtures.authRegAppImage, fixtures.authRegAuthConfig = buildAuthRegistryImage(
 		t,
 		someAppName,
 		filepath.Join("testdata", "analyzer", "app-image"),
+		testRegistry,
 		"--build-arg", "fromImage="+containerBaseImage,
 		"--build-arg", "metadata="+appMeta,
 	)
-	fixtures.readOnlyRegAppImage = readOnlyRegistry.RepoName(someAppName)
+	testRegistry.Add(someAppName, ih.NewImagePrivileges(readWriteImage))
 
 	someOtherAppName := "some-other-app-image-" + h.RandString(10)
 	fixtures.authRegOtherAppImage, _ = buildAuthRegistryImage(
 		t,
 		someOtherAppName,
 		filepath.Join("testdata", "analyzer", "app-image"),
+		testRegistry,
 		"--build-arg", "fromImage="+containerBaseImage,
 		"--build-arg", "metadata="+appMeta,
 	)
+	testRegistry.Add(someOtherAppName, ih.NewImagePrivileges(readWriteImage))
 
-	someCacheName := "some-cache-image-" + h.RandString(10)
+	// setup read only images
+	someReadOnlyAppName := "some-readonly-app-image-" + h.RandString(10)
+	fixtures.readOnlyRegAppImage, _ = buildAuthRegistryImage(
+		t,
+		someReadOnlyAppName,
+		filepath.Join("testdata", "analyzer", "app-image"),
+		testRegistry,
+		"--build-arg", "fromImage="+containerBaseImage,
+		"--build-arg", "metadata="+appMeta,
+	)
+	testRegistry.Add(someReadOnlyAppName, ih.NewImagePrivileges(onlyReadImage))
+
+	someCacheName := "some-cache-image"
 	fixtures.authRegCacheImage, _ = buildAuthRegistryImage(
 		t,
 		someCacheName,
 		filepath.Join("testdata", "analyzer", "cache-image"),
+		testRegistry,
 		"--build-arg", "fromImage="+containerBaseImage,
 		"--build-arg", "metadata="+cacheMeta,
 	)
-	fixtures.readOnlyRegCacheImage = readOnlyRegistry.RepoName(someCacheName)
+	testRegistry.Add(someCacheName, ih.NewImagePrivileges(onlyReadImage))
+	fixtures.readOnlyRegCacheImage = testRegistry.RepoName(someCacheName)
+
+	// setup no access image
+	inaccessibleImage := "inaccessible-image"
+	testRegistry.Add(inaccessibleImage, ih.NewImagePrivileges(noAccessImage))
+	fixtures.inaccessibleImage = testRegistry.RepoName(inaccessibleImage)
```

Comment:
Why is this needed?

Reply:
(it seems like a no-op, except for setting image name on `fixtures`)

Reply:
Yes, this one is just creating an image without reading or writing access that is being used in some test cases

Reply:
Do we need it? What happens if this is left out?

Reply:
We can remove them, but I will need to send a different image on some test cases, the ones where I am validating read or read/write access. I can try to do it, let me check

Reply:
I will need to create at least one image with only writable permissions, there is a test in line 960 where I need an image that is not readable, we can accomplish that with a only writable image or an image without any permissions like this one.

Reply:
I think I understand how this is working now (see https://github.com/buildpacks/lifecycle/pull/685#discussion_r686323993). This seems fine to me.